File: Task-03/C++/subtask4.cpp
//sub-task 4  AM@FOSS
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
    int n;

    // Open input file for reading
    ifstream inputFile("input.txt");

    // Read the number from the input file
    inputFile >> n;
    inputFile.close();

    // Open output file for writing
    ofstream outputFile("output.txt");
    // Generate and write the diamond pattern to the output file
    for (int i = 1; i <= n; i += 2) {
        outputFile << string((n - i) / 2, ' ') << string(i, '*') << '\n';
    }

    for (int i = n - 2; i >= 1; i -= 2) {
        outputFile << string((n - i) / 2, ' ') << string(i, '*') << '\n';
    }

    outputFile.close();

    return 0;
}

File: Task-03/python/subtask4.py
#Sub-Task 4: Write a program that reads a number n from a file named input.txt and generates a diamond pattern of asterisks (*) in another file named output.txt.

File=open("input.txt","r")
File2=open("output.txt","w")  #Opening the files
data=File.read()
data=int(data)
for k in range(1, data+1, 2):
    File2.write(" " * ((data-k) // 2) + "*" * k + "\n")
 
for k in range(data-2, 0, -2):
    File2.write(" " * ((data-k) // 2) + "*" * k +"\n")
print("Task Completed")
File.close()
File2.close()   #closing files to avoid error
